invoke dynamic bug for equals/hashcode/toString of Record classes

When compiling the invokedynamic found in the equals/hashcode/toString methods of a record. Here's a trivial reproducer:
```
class RecordTest {
    static record R (String name, Integer age) {}
    
    public static void main(String[] args) {
        R r = new R(args[0], args.length);
    }
}
```

The byte codes are as one might expect:
```
  public final boolean equals(java.lang.Object);
    Code:
       0: aload_0
       1: aload_1
       2: invokedynamic #25,  0             // InvokeDynamic #0:equals:(LRecordTest$R;Ljava/lang/Object;)Z
       7: ireturn

```

When compiling, qbicc reports:
```
RecordTest.java:4: warning: Failed to invoke bootstrap method handle: (interpreter) java.lang.BootstrapMethodError: bootstrap method initialization exception
    member: method equals@ bci 2
  location: type RecordTest$R
```

I've done a little digging in the debugger, and we're getting a `WrongMethodTypeException`.

--- a/compiler/src/main/java/org/qbicc/type/definition/classfile/MethodParser.java
+++ b/compiler/src/main/java/org/qbicc/type/definition/classfile/MethodParser.java
@@ -1668,7 +1668,6 @@ final class MethodParser {
                             // 1. allocate the array
                             int bootstrapArgCnt = getClassFile().getBootstrapMethodArgumentCount(bootstrapMethodIdx);
                             List<ParameterElement> targetParameters = targetMethod.getParameters();
-                            boolean vararg = targetParameters.size() == 4 && targetMethod.isVarargs() && !targetMethod.isSignaturePolymorphic();
                             TypeParameterContext tpc;
                             ExecutableElement rootElement = gf.getRootElement();
                             if (rootElement instanceof TypeParameterContext) {
@@ -1683,6 +1682,27 @@ final class MethodParser {
                                 int argIdx = getClassFile().getBootstrapMethodArgumentConstantIndex(bootstrapMethodIdx, i);
                                 if (argIdx != 0) {
                                     argsArray[i] = vm.box(ctxt, getClassFile().getConstantValue(argIdx, tpc));
+                                }
+                            }
+                            // 1.5 If the target method is a normal varargs method and the bootstrap args represent
+                            //     a mix of normal and trailing arguments, bundle the trailing arguments in an array
+                            //     See JVM Spec 5.4.3.6, task 2, step 2.
+                            if (bootstrapArgCnt > 0 && targetMethod.isVarargs() && !targetMethod.isSignaturePolymorphic()) {
+                                int normalStaticArgs = targetParameters.size() - 3 - 1;
+                                int trailingStaticArgs = bootstrapArgCnt - normalStaticArgs;
+                                if (trailingStaticArgs > 0 && trailingStaticArgs != bootstrapArgCnt) {
+                                    VmObject[] adjustedArgs = new VmObject[normalStaticArgs+1];
+                                    for (int i=0; i<normalStaticArgs; i++) {
+                                        adjustedArgs[i] = argsArray[i];
+                                    }
+                                    VmObject[] trailingArgs = new VmObject[trailingStaticArgs];
+                                    for (int i=normalStaticArgs; i<argsArray.length; i++) {
+                                        trailingArgs[i-normalStaticArgs] = argsArray[i];
+                                    }
+                                    TypeDescriptor elementDescriptor = ((ArrayTypeDescriptor) targetParameters.get(targetParameters.size()-1).getTypeDescriptor()).getElementTypeDescriptor();
+                                    ObjectType elemType = (ObjectType)ctxt.resolveTypeFromDescriptor(elementDescriptor, tpc, TypeSignature.synthesize(ctxt, elementDescriptor));
+                                    adjustedArgs[adjustedArgs.length-1] = vm.newArrayOf(elemType.getDefinition().load().getVmClass(), trailingArgs);
+                                    args = vm.newArrayOf(objectClass, adjustedArgs);
                                 }
                             }
                             VmReferenceArray appendixResult = vm.newArrayOf(objectClass, 1);

